test/built-ins/RegExp/named-groups/non-unicode-property-names.js

```diff
@@ -0,0 +1,45 @@
+// Copyright 2017 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Exotic named group names in non-Unicode RegExps
+esid: pending
+features: [regexp-named-groups]
+includes: [compareArray.js]
+---*/
+
+assert.sameValue("a", /(?<π>a)/.exec("bab").groups.π);
+assert.throws(SyntaxError, () => eval('/(?<\\u{03C0}>a)/'), "\\u{} escapes allowed only in Unicode mode");
+assert.sameValue("a", /(?<π>a)/.exec("bab").groups.\u03C0);
+assert.sameValue("a", /(?<$>a)/.exec("bab").groups.$);
+assert.sameValue("a", /(?<_>a)/.exec("bab").groups._);
+assert.sameValue("a", /(?<$𐒤>a)/.exec("bab").groups.$𐒤);
```

Comment:
This should be an early syntax error, because 𐒤 is parsed as two separate code units in non-Unicode mode. 

https://tc39.github.io/ecma262/#sec-pattern-semantics
> If a BMP pattern contains a non-BMP SourceCharacter the entire pattern is encoded using UTF-16 and the individual code units of that encoding are used as the elements of the List.